src/mono/mono/mini/intrinsics.c

```diff
@@ -2045,16 +2041,16 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 	}
 
 #ifdef MONO_ARCH_SIMD_INTRINSICS
-#ifndef TARGET_WASM
 	if (cfg->opt & MONO_OPT_SIMD) {
-#endif
 		ins = mono_emit_simd_intrinsics (cfg, cmethod, fsig, args);
```

Comment:
An alternative would be to move the if (cfg->opt check to mono_emit_simd_intrinsics ()).